systemverilog-plugin/UhdmAst.cc

```diff
@@ -503,10 +500,16 @@ static void convert_packed_unpacked_range(AST::AstNode *wire_node)
     }();
     if (convert_node) {
         if (wire_node->multirange_dimensions.empty()) {
-            packed_size = add_multirange_attribute(wire_node, packed_ranges);
-            unpacked_size = add_multirange_attribute(wire_node, unpacked_ranges);
-            size = packed_size * unpacked_size;
-            ranges.push_back(make_range(size - 1, 0));
+            size_t packed_size = add_multirange_attribute(wire_node, packed_ranges);
+            size_t unpacked_size = add_multirange_attribute(wire_node, unpacked_ranges);
+            if (packed_ranges.size() == 1 && unpacked_ranges.empty()) {
+                ranges.push_back(packed_ranges[0]->clone());
+            } else if (unpacked_ranges.size() == 1 && packed_ranges.empty()) {
+                ranges.push_back(unpacked_ranges[0]->clone());
+            } else {
+                size_t size = packed_size * unpacked_size;
+                ranges.push_back(make_range(size - 1, 0));
```

Comment:
``add_multirange_attribute`` is returning actual size of wire, so it should always return at least ``1``, but you are right, this can be problematic if ``size`` is ``0``. I've added assert to make sure it is always positive.

New run: https://github.com/antmicro/yosys-systemverilog/actions/runs/4281712305

Reply:
@wsipak: the `add_multirange_attribute` does `(max - min) + 1` for every range, and returns 1 when there are no ranges, so 0 is not possible without overflow. https://github.com/chipsalliance/yosys-f4pga-plugins/blob/f5e5cab4550ce18f707ce2e26dea24890ae0732e/systemverilog-plugin/UhdmAst.cc#L239-L240
But the assert is OK, just in case the function is changed.